src/thunderbots/software/grsim_communication/grsim_backend.cpp

```diff
@@ -55,25 +60,39 @@ void GrSimBackend::sendPrimitives(
                 GrsimCommandPrimitiveVisitor(robot, ball);
             prim->accept(grsim_command_primitive_visitor);
 
-            MotionController::MotionControllerCommand motion_controller_command =
-                grsim_command_primitive_visitor.getMotionControllerCommand();
+            std::variant<MotionController::PositionCommand,
+                         MotionController::VelocityCommand>
+                motion_controller_command =
+                    grsim_command_primitive_visitor.getMotionControllerCommand();
 
             MotionController::Velocity robot_velocities =
-                MotionController::bangBangVelocityController(
-                    robot, motion_controller_command.global_destination,
-                    motion_controller_command.final_speed_at_destination,
-                    motion_controller_command.final_orientation, delta_time.count(),
-                    ROBOT_MAX_SPEED_METERS_PER_SECOND, ROBOT_MAX_ANG_SPEED_RAD_PER_SECOND,
-                    ROBOT_MAX_ACCELERATION_METERS_PER_SECOND_SQUARED,
-                    ROBOT_MAX_ANG_ACCELERATION_RAD_PER_SECOND_SQUARED);
+                motionController.bangBangVelocityController(robot, delta_time.count(),
+                                                            motion_controller_command);
+
+            double kick_speed_meters_per_second;
+            bool chip_instead_of_kick;
+            bool dribbler_on;
+
+            if (auto val = std::get_if<MotionController::PositionCommand>(
+                    &motion_controller_command))
+            {
+                kick_speed_meters_per_second = val->kick_speed_meters_per_second;
+                chip_instead_of_kick         = val->chip_instead_of_kick;
+                dribbler_on                  = val->dribbler_on;
+            }
+            else if (auto val = std::get_if<MotionController::VelocityCommand>(
+                         &motion_controller_command))
+            {
+                kick_speed_meters_per_second = val->kick_speed_meters_per_second;
+                chip_instead_of_kick         = val->chip_instead_of_kick;
+                dribbler_on                  = val->dribbler_on;
+            }
```

Comment:
Could we use the same `std::visit` expression we used below here, just to be consistent?